soc/arm/atmel_sam/common/atmel_sam_dt.h

```diff
@@ -43,4 +45,17 @@
 		ATMEL_SAM_PIN_PERIPH(inst, idx) << 16		\
 	}
 
+/* Devicetree related macros related to clock */
+#ifndef CONFIG_SOC_ATMEL_SAM_MDIV
+#define CONFIG_SOC_ATMEL_SAM_MDIV 1
+#endif
```

Comment:
If we are going to refactor clock, we should do it properly too (i.e. clock generator node in DT + each device referencing the node with proper bus/div params).

For example, see NXP (SIM) and ST (RCC) device tree:
https://github.com/zephyrproject-rtos/zephyr/blob/e7c7f911a9cd99768c9051b30c2f206e9998de1c/dts/arm/nxp/nxp_k6x.dtsi#L135-L141

https://github.com/zephyrproject-rtos/zephyr/blob/e7c7f911a9cd99768c9051b30c2f206e9998de1c/dts/arm/nxp/nxp_k6x.dtsi#L169

In case of Atmel SAM series, this would be called `ckgr`.

Reply:
Hi @stephanosio, the idea here is only do a simple clean up and moving from SoC about all DT_ related to new file atmel_sam_dt.h. I don't have any intention to touch on clocks right now.
This is the only one and now we can assume on Atmel SAM SoC all DT related things have this file as entry point.